buildpacks/java/java/func_yaml.go

```diff
@@ -34,14 +35,33 @@ func ParseFuncYaml(filedir string, logger bard.Logger) FuncYaml {
 
 	options := optionsToMap(cfg.Options, logger)
 	envs := envsToMap(cfg.Envs, logger)
-
+	name := getName(cfg.Envs, cfg.Name)
 	return FuncYaml{
+		Name:    name,
 		Options: options,
 		Envs:    envs,
 		Exists:  true,
 	}
 }
 
+func getName(envs knfn.Envs, nameFromYaml string) string {
+	result := nameFromYaml
+
+	if envs == nil {
+		return result
+	}
+
+	for _, e := range envs {
+		key := *e.Name
+		if key == "FUNCTION_NAME" {
```

Comment:
Java doesn't make use of this environment variable, only the python invoker does